### PR TITLE
polish(pu): polish comments and styles in files within torch_utils/network

### PR DESCRIPTION
**Pull request by puyuan1996**
## Description
 - polish the comments and code styles in the files located within the ding/torch_utils/network/ directory.

## Check List

- [x] merge the latest version source branch/repo, and resolve all the conflicts
- [x] pass style check
- [x] pass all the tests

### CODE REVIEW

File: ding/torch_utils/network/activation.py [hunk 1]
```diff
@@ -17,24 +28,22 @@ def forward(self, x):
 class GLU(nn.Module):
     """
     Overview:
-        Gating Linear Unit.
+        Gating Linear Unit (GLU), a specific type of activation function. This module also supports 2D convolution,
+         in which case, the input and context must have the same shape.
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:32]: polish indent

File: ding/torch_utils/network/activation.py [hunk 2]
```diff
@@ -62,20 +71,36 @@ def forward(self, x: torch.Tensor, context: torch.Tensor) -> torch.Tensor:
...
     def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Overview:
+            Compute the Swish transformation of the input tensor.
+        Arguments:
+            - x (:obj:`torch.Tensor`): The input tensor.
+        Returns:
+            - x (:obj:`torch.Tensor`): The output tensor after Swish transformation.
+        """
         x = x * torch.sigmoid(x)
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:95]: directly return

File: ding/torch_utils/network/activation.py [hunk 3]
```diff
@@ -17,24 +28,22 @@ def forward(self, x):
 class GLU(nn.Module):
     """
     Overview:
-        Gating Linear Unit.
+        Gating Linear Unit (GLU), a specific type of activation function. This module also supports 2D convolution,
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:31]: add paper or reference link

File: ding/torch_utils/network/activation.py [hunk 4]
```diff
@@ -62,20 +71,36 @@ def forward(self, x: torch.Tensor, context: torch.Tensor) -> torch.Tensor:
 
 
 class Swish(nn.Module):
+    """
+    Overview:
+        Swish activation function, which is a smooth, non-monotonic activation function.
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:76]: add paper or reference link

File: ding/torch_utils/network/activation.py [hunk 5]
```diff
@@ -84,19 +109,27 @@ def __init__(self):
...
 def build_activation(activation: str, inplace: bool = None) -> nn.Module:
     """
     Overview:
-        Return the activation module according to the given type.
+        Build and return the activation module according to the given type.
     Arguments:
-        - activation (:obj:`str`): The type of activation module, now supports \
-            ['relu', 'glu', 'prelu', 'swish', 'gelu', 'tanh', 'sigmoid', 'softplus', 'elu', 'square', 'identity'].
-        - inplace (:obj:`bool`): Execute the operation in-place in activation, defaults to ``None``.
+        - activation (:obj:str): The type of activation module, now supports
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:128]: Youn need to add an newline character

Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:128]: :obj:\`str\`

File: ding/torch_utils/network/activation.py [hunk 6]
```diff
@@ -84,19 +109,27 @@ def __init__(self):
...
+        ['relu', 'glu', 'prelu', 'swish', 'gelu', 'tanh', 'sigmoid', 'softplus', 'elu', 'square', 'identity'].
+        - inplace (:obj:bool, optional): Execute the operation in-place in activation, defaults to None.
```
Comment (PaParaZz1) [inline ding/torch_utils/network/activation.py:130]: Optional[bool]

File: ding/torch_utils/network/gumbel_softmax.py [hunk 7]
```diff
@@ -4,44 +4,53 @@
...
-    .. note:
-
-        For more gumbelsoftmax info, you can refer to
-        the paper <https://arxiv.org/abs/1611.01144>
+        An `nn.Module` that computes GumbelSoftmax.
+    Interfaces:
+        __init__, forward, gumbel_softmax_sample
 
+    .. note::
+        For more information on GumbelSoftmax, refer to the paper [Categorical
```
Comment (PaParaZz1) [inline ding/torch_utils/network/gumbel_softmax.py:14]: it is not a proper newline

File: ding/torch_utils/network/gumbel_softmax.py [hunk 8]
```diff
@@ -4,44 +4,53 @@
...
+            - x (:obj:`torch.Tensor`): Unnormalized log-probabilities.
+            - temperature (:obj:`float`): Non-negative scalar controlling the sharpness of the distribution.
+            - hard (:obj:`bool`): If `True`, returns one-hot encoded labels. Default is `False`.
         Returns:
-            - output (:obj:`torch.Tensor`): sample from GumbelSoftmax distribution
+            - output (:obj:`torch.Tensor`): Sample from Gumbel-Softmax distribution.
         Shapes:
-            - x: :math:`(B, N)`, while B is the batch size, N is number of classes
-            - output: :math:`(B, N)`, while B is the batch size, N is number of classes
+            - Input, x: :math:`(B, N)`, where `B` is the batch size and `N` is the number of classes.
```
Comment (PaParaZz1) [inline ding/torch_utils/network/gumbel_softmax.py:52]: why input here

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/torch_utils/network/gtrxl.py [hunk 9]
```diff
@@ -44,16 +47,16 @@ def forward(self, pos_seq: torch.Tensor):
...
         """
         Arguments:
-            - input_dim: (:obj:`int`): dimension of input.
-            - bg (:obj:`bg`): gate bias. By setting bg > 0 we can explicitly initialize the gating mechanism to
-            be close to the identity map. This can greatly improve the learning speed and stability since it
-            initializes the agent close to a Markovian policy (ignore attention at the beginning).
+            - input_dim (:obj:`int`): The dimensionality of the input.
+            - bg (:obj:`bg`): The gate bias. By setting bg > 0 we can explicitly initialize the gating mechanism to
+                be close to the identity map. This can greatly improve the learning speed and stability since it
+                initializes the agent close to a Markovian policy (ignore attention at the beginning).
```
Comment (HarryXuancy) [inline ding/torch_utils/network/gtrxl.py:59]: new line character "\\"

File: ding/torch_utils/network/gtrxl.py [hunk 10]
```diff
@@ -206,28 +220,30 @@ def __init__(self, input_dim: int, head_dim: int, head_num: int, dropout: nn.Mod
...
+
         .. note::
             See the following material for better understanding:
                 https://github.com/kimiyoung/transformer-xl/issues/8
                 https://arxiv.org/pdf/1901.02860.pdf (Appendix B)
         Arguments:
-            - x (:obj:`torch.Tensor`): input tensor of shape (cur_seq, full_seq, bs, head_num).
-            - zero_upper (:obj:`bool`): if True set the upper-right triangle to zero.
+           - x (:obj:`torch.Tensor`): The input tensor with shape (cur_seq, full_seq, bs, head_num).
+            - zero_upper (:obj:`bool`): If True, the upper-right triangle of the matrix is set to zero.
```
Comment (HarryXuancy) [inline ding/torch_utils/network/gtrxl.py:243]: The two lines are not aligned

File: ding/torch_utils/network/gtrxl.py [hunk 11]
```diff
@@ -498,16 +522,16 @@ def get_memory(self):
...
-            - x (:obj:`torch.Tensor`): input tensor. Shape (seq_len, bs, input_size).
-            - batch_first (:obj:`bool`): if the input data has shape (bs, seq_len, input_size), set this param to \
-                ``True`` in order to transpose along the first and second dimension and obtain shape \
-                (seq_len, bs, input_size). This param doesn't affects the output memory.
-            - return_mem (:obj:`bool`): if this param is False, return only the output tensor without dict.
+           - x (:obj:`torch.Tensor`): The input tensor with shape (seq_len, bs, input_size).
+            - batch_first (:obj:`bool`, optional): If the input data has shape (bs, seq_len, input_size),
+                set this parameter to True to transpose along the first and second dimension and obtain shape
+                (seq_len, bs, input_size). This does not affect the output memory. Default is False.
+            - return_mem (:obj:`bool`, optional): If False, return only the output tensor without dict. Default is True.
```
Comment (HarryXuancy) [inline ding/torch_utils/network/gtrxl.py:531]: The last two arguments are not aligned to the first

File: ding/torch_utils/network/gtrxl.py [hunk 12]
```diff
@@ -394,10 +416,8 @@ def forward(
 class GTrXL(nn.Module):
     """
     Overview:
-        GTrXL Transformer implementation.
-
-    .. note::
-        For details refer to Stabilizing Transformer for Reinforcement Learning: https://arxiv.org/abs/1910.06764
+        GTrXL Transformer implementation as described in "Stabilizing Transformer for Reinforcement Learning"
+        (https://arxiv.org/abs/1910.06764).
```
Comment (HarryXuancy) [inline ding/torch_utils/network/gtrxl.py:420]: Maybe we can add a brief description at the beginning of this python file, like "This file implements the GTrXL Transformer as described in "Stabilizing Transformer for Reinforcement Learning" (https://arxiv.org/abs/1910.06764)."

File: ding/torch_utils/network/rnn.py [hunk 13]
```diff
@@ -124,15 +130,15 @@ def _after_forward(self,
...
 
 class LSTM(nn.Module, LSTMForwardWrapper):
-    r"""
+    """
     Overview:
-        Implimentation of LSTM cell with LN
+        Implementation of an LSTM cell with Layer Normalization (LN).
     Interface:
-        forward
+        forward.
```
Comment (HarryXuancy) [inline ding/torch_utils/network/rnn.py:137]: why . here, other interface have no .

File: ding/torch_utils/network/rnn.py [hunk 14]
```diff
@@ -265,32 +267,47 @@ def forward(self,
 
 
 class GRU(nn.GRUCell, LSTMForwardWrapper):
-    r"""
-    Overview:
-        Wrap the torch.nn.GRUCell, format the input and output
+    """
+    This class extends the `torch.nn.GRUCell` and `LSTMForwardWrapper` classes, and formats inputs and outputs
+        accordingly.
```
Comment (HarryXuancy) [inline ding/torch_utils/network/rnn.py:272]: miss Overview

File: ding/torch_utils/network/nn_module.py [hunk 15]
```diff
@@ -449,17 +486,17 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
...
-        - num (:obj:`int`): number of states of the one hot encoding
-        - num_first (:obj:`bool`): If ``num_first`` is False, the one hot encoding is added as the last; \
-            Otherwise as the first dimension.
+        - val (:obj:`torch.LongTensor`): Each element contains the state to be encoded, the range should be [0, num-1]
+        - num (:obj:`int`): Number of states of the one-hot encoding
+        - num_first (:obj:`bool`, optional): If False, the one-hot encoding is added as the last dimension; otherwise,
+            it is added as the first dimension. Default is False.
     Returns:
-        - one_hot (:obj:`torch.FloatTensor`)
+        - one_hot (:obj:`torch.FloatTensor`): The one-hot encoded tensor
```
Comment (HarryXuancy) [inline ding/torch_utils/network/nn_module.py:499]: missing the .

File: ding/torch_utils/network/resnet.py [hunk 16]
```diff
@@ -204,26 +403,59 @@ def forward(self, x):
 
 
 class Bottleneck(nn.Module):
+    """
+    Overview:
+        The Bottleneck class is a basic block used to build ResNet networks. It is a part of the PyTorch's
+        implementation of ResNet. This block is designed with several layers including a convolutional layer,
```
Comment (HarryXuancy) [inline ding/torch_utils/network/resnet.py:409]: new line character

File: ding/torch_utils/network/soft_argmax.py [hunk 17]
```diff
@@ -4,39 +4,36 @@
 
 
 class SoftArgmax(nn.Module):
-    r"""
+    """
     Overview:
-        An nn.Module that computes SoftArgmax
+        An nn.Module that computes the SoftArgmax operation.
```
Comment (HarryXuancy) [inline ding/torch_utils/network/soft_argmax.py:9]: Maybe a description that this softmax is 2-dimension softmax is better.

File: ding/torch_utils/network/res_block.py [hunk 18]
```diff
@@ -39,11 +39,11 @@ def __init__(
             Init the 2D convolution residual block.
```
[the remark this answers is not in the document]
Comment (puyuan1996) [reply ding/torch_utils/network/res_block.py:39]: ok

File: ding/torch_utils/network/soft_argmax.py (hunk 17 above)
Comment (puyuan1996) [reply ding/torch_utils/network/soft_argmax.py:9]: polished now

**Review by PaParaZz1 — APPROVED** [review]
